test/object-store/sync/app.cpp

```diff
@@ -2435,6 +2440,9 @@ TEST_CASE("app: sync integration", "[sync][app]") {
 
         auto redir_transport = std::make_shared<HookedTransport>();
         auto redir_provider = std::make_shared<HookedSocketProvider>(logger, "");
+        std::mutex logout_mutex;
```

Comment:
These tests are purely synchronous and all the waiting on CVs isn't doing anything.

Reply:
It's true that the first set of `App` redirect tests are purely synchronous, but the App calls for the websocket redirect tests under the "Test websocket redirect with existing session" section are being driven by the event loop due to redirect responses from websocket connect, so I will still need the mutex and CV. In fact, we've had some occasional failures where the `wait_for_download()` returned before the user was logged out.